gwctl/CODE_ORGANIZATION.md

```diff
@@ -0,0 +1,80 @@
+                                                                                                                 |
+                                                                                                                 |
+                +------------+                           +--------------+                               TargetRef is Namespace
+                |   Policy   |                      +----> GatewayClass <-------+                                |
+                +----+-------+                      |    +--------------+       |                                |
+                     |                              |                           |            +---------+         |
+                     |                              |                           |  +-------->|Namespace|<--------+
+                     |                      Parent GatewayClass                 |  |         +---------+
+                     |                              |                           |  |
+                     |                              |                           |  |
+                     |                              |                           |  |
+                     |                        +-----+-----+               +-----+--+--+
+            TargetRef is HTTPRoute    +------->  Gateway  <---+           |  Gateway  <--+
+                     |                |       +-----------+   |           +-----------+  |
+                     |                |                       |                          |
+                     |                |                       |                          |
+                     |           Attached to             Attached to                  Attached to
+                     |                |                       |                          |
+                     |                |                       |                          |
+                     |           +----+------+             +--+--------+            +----+------+
+                     +---------->| HTTPRoute |             | HTTPRoute |            | HTTPRoute |
+                                 +----+------+             +----+------+            +-----+-----+
+                                      |                         |                         |
+                                      |                         |                         |
+                                  Routes to                 Routes to                 Routes to
+                                      |                         |                         |
+                                      |       +---------+       |                    +----v----+
+                                      +-------> Backend <-------+                    | Backend |
+                                              +---------+                            +---------+
+            ```
```

Comment:
Non-blocking - this _might_ be easier to represent with a [Mermaid.js diagram](https://mermaid.js.org/).